common/src/main/scala/net/psforever/objects/equipment/JammingUnit.scala

```diff
@@ -0,0 +1,271 @@
+    * Due to considerations of the object that is the target, this is left to be implemented by a subclass.
+    * We merely stop the timer.
+    * @param target an object that can be affected by the jammered status
+    */
+  def CancelJammeredStatus(target : Any) : Unit = {
+    JammableObject.Jammed = false
+    jammeredStatusTimer.cancel
+  }
+
+  val jammableBehavior : Receive = {
+    case JammableUnit.Jammered(cause) =>
+      TryJammerEffectActivate(JammableObject, cause)
+
+    case JammableUnit.ClearJammeredSound() =>
+      CancelJammeredSound(JammableObject)
+
+    case JammableUnit.ClearJammeredStatus() =>
+      CancelJammeredStatus(JammableObject)
+  }
+}
+
+/**
+  * A common mix-in variation to manage common responses to the "jammerable" status for game objects with mounted weapons.
+  * @see `MountedWeapons`
+  * @see `Service`
+  * @see `VehicleAction`
+  * @see `VehicleService`
+  * @see `VehicleServiceMessage`
+  * @see `Zone.VehicleEvents`
+  */
+trait JammableMountedWeapons extends JammableBehavior {
+  _ : Actor =>
+
+  override def StartJammeredSound(target : Any, dur : Int) : Unit = {
+    target match {
+      case obj : PlanetSideServerObject with MountedWeapons with JammableUnit if !jammedSound =>
+        obj.Zone.VehicleEvents ! VehicleServiceMessage(obj.Zone.Id, VehicleAction.PlanetsideAttribute(Service.defaultPlayerGUID, obj.GUID, 27, 1))
+        super.StartJammeredSound(target, dur)
+      case _ => ;
+    }
+  }
+
+  override def StartJammeredStatus(target : Any, dur : Int) : Unit = {
+    target match {
+      case obj : PlanetSideServerObject with MountedWeapons with JammableUnit if !obj.Jammed =>
+        JammableMountedWeapons.JammeredStatus(obj, 1)
+        super.StartJammeredStatus(target, dur)
+      case _ => ;
+    }
+  }
+
+  override def CancelJammeredSound(target : Any) : Unit = {
+    target match {
+      case obj : PlanetSideServerObject if jammedSound =>
+        obj.Zone.VehicleEvents ! VehicleServiceMessage(obj.Zone.Id, VehicleAction.PlanetsideAttribute(Service.defaultPlayerGUID, obj.GUID, 27, 0))
+      case _ => ;
+    }
+    super.CancelJammeredSound(target)
+  }
+
+  override def CancelJammeredStatus(target : Any) : Unit = {
+    target match {
+      case obj : PlanetSideServerObject with MountedWeapons with JammableUnit if obj.Jammed =>
+        JammableMountedWeapons.JammeredStatus(obj, 0)
+      case _ => ;
+    }
+    super.CancelJammeredStatus(target)
+  }
+}
+
+object JammableMountedWeapons {
+  /**
+    * Retrieve all of the weapons on a `MountedWeapons` target object and apply a jammered status effect to each.
+    * @param target an object that can be affected by the jammered status
+    * @param statusCode the jammered status condition;
+    *                   0 for deactivation;
+    *                   1 for activation
+    */
+  def JammeredStatus(target : PlanetSideServerObject with MountedWeapons, statusCode : Int) : Unit = {
```

Comment:
Is statusCode not better as a Boolean, or are other statuses possible?

Reply:
`statusCode` gets fed into a `PAM`.  I see no reason to cast from `Boolean` when the value that is needed can directly be a `Long`.

If some reason to ever run an unscheduled `JammeredStatus(obj, obj.Jammed)` arises, I'll consider it useful.

Reply:
👍